src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -211,6 +77,9 @@ public static void ThrowException(ref JSMarshalerArgument arg)
         {
             arg.ToManaged(out Exception? ex);
 
+#if FEATURE_WASM_THREADS
+            JSProxyContext.PopOperation();
```

Comment:
This method should probably have a xmldoc comment specifying that it pops the current operation, since most method calls would not unbalance the operation stack in this way